### PR TITLE
addding doc on deploying policies with GitOps

### PR DESCRIPTION
**Pull request by dockerymick**
https://github.com/open-cluster-management/backlog/issues/10780

### CODE REVIEW

File: risk_compliance/deploy_gitops.adoc [hunk 1]
```diff
@@ -0,0 +1,177 @@
...
++
+*Tips*:
+
+* It is recommend to remove the policies you are not planning to use.
+
+* Spend some time going through the policies to make sure you understand the following criteria for each policy:
+
+** Purpose: Understand what the policy does.
+
+** Remediation Action: Does the policy only inform you of compliance, or enforce the policy and make changes? Look at spec.remediationAction. If changes are enforced, make sure you understand the functional expectation. Remember to check which policies support enforcement. For more information, view the Validate section.
```
Comment (dhaiducek) [inline risk_compliance/deploy_gitops.adoc]: I think this should have code formatting:
spec.remediationAction --> \`spec.remediationAction\`

File: risk_compliance/deploy_gitops.adoc [hunk 2]
```diff
@@ -0,0 +1,177 @@
...
++
+*Note*: The `spec.remediationAction` set for the policy overrides any remediation action that is set in the individual `spec.policy-templates`.
+
+** Placement: What clusters is the policy deployed to? By default, most policies target the clusters with the environment: dev label. Some policies may target OpenShift clusters or another label. You can update or add additional labels to include other clusters. When there is no specific value, the policy is applied to all of your clusters. You can also create multiple copies of a policy and customize each one if you want to use a policy configured one way for one set of clusters and configured another way for another set of clusters.
```
Comment (dhaiducek) [inline risk_compliance/deploy_gitops.adoc]: Code block: \`environment: dev\`

File: risk_compliance/deploy_gitops.adoc [hunk 3]
```diff
@@ -0,0 +1,177 @@
...
+git add my-policies/
+----
+
+. Commit the changes and customize the message as desired. Run the following command:
++
+----
+git commit -m “Policies to deploy to the hub cluster”
+----
+
+. Push the changes to the main default branch of your forked Git repository’s that is used for GitOps. Run the following command:
```
Comment (dhaiducek) [inline risk_compliance/deploy_gitops.adoc]: It doesn't actually have to be the `main` branch or the default branch of the repo, so we can probably rephrase:
> ...to the main default branch of your...

to:

> ...to the branch of your...

File: risk_compliance/deploy_gitops.adoc [hunk 4]
```diff
@@ -0,0 +1,177 @@
...
+----
+
+Your changes are committed.
+
+[#deploying-policies-to-your-cluster]
+== Deploying policies to your cluster
+
+After you push your changes, you can deploy the policies to your {product-title} installation. Post deployment, your hub cluster is conncted to your Git repository. Any further changes to your chosen branch of the Git repository is reflected in your cluster.
+
+The `deploy.sh` script creates `Channel` and `Subscription` resources in your hub cluster. The channel connects to the Git repository, and the subscription specifies the data to bring to the cluster through the channel. In this way, all policies defined in the specified subdirectory is created on your hub. After the policies are created by the subscription, {product-title-short} analyzes the policies and creates additional policy resources in the namespace associated with each managed cluster that the policy is applied to, based on the defined placement rule.
```
Comment (dhaiducek) [inline risk_compliance/deploy_gitops.adoc]: "subdirectory is created" --> "subdirectory are created"

File: risk_compliance/deploy_gitops.adoc [hunk 5]
```diff
@@ -0,0 +1,177 @@
...
+cp -R community/* my-policies/
+
+cp -R stable/* my-policies/
+----
++
+Now that you have all of the policies in a single parent directory structure, you can make edits to the policies in your fork.
++
+*Tips*:
+
+* It is recommend to remove the policies you are not planning to use.
```
Comment (swopebe) [inline risk_compliance/deploy_gitops.adoc]: Let's chat about this set up, since we should not recommend. Usually we would use **Best practice:**

File: risk_compliance/deploy_gitops.adoc [hunk 6]
```diff
@@ -0,0 +1,177 @@
+[#deploy-policies-using-gitops]
+= Deploy policies using GitOps
+
+The governance framework allows you to deploy a set of policies across a fleet of managed clusters. You can leverage the open source community, https://github.com/open-cluster-management/policy-collection[`policy-collection`] by contributing and using the policies in the repository. Policies in each of the `stable` and `community` folders from the open source community are further organized according to link:https://nvd.nist.gov/800-53/Rev4[NIST Special Publication 800-53]. 
+
+Continue reading to learn how to use GitOps to automate and track policy updates and creation through a Git repository.
+
+*Prerequsite*: Before you begin, be sure to create you own fork of the `policy-collection` repository.
```
Comment (swopebe) [inline risk_compliance/deploy_gitops.adoc]: Let's make this active: "...fork the ___ repository."

File: risk_compliance/deploy_gitops.adoc [hunk 7]
```diff
@@ -0,0 +1,177 @@
...
+. Make sure that your command line interface (CLI) is configured to create resources on the correct cluster with the following command:
++
+----
+oc cluster-info
+----
++
+The output of the command displays the API server details for the cluster, where {product-title-short} is installed. If the correct URL is not displayed, configure your CLI to point to the correct cluster. 
+
+//^is there a reference that we can add to help the
+// user configure the CLI?
```
Comment (dhaiducek) [inline risk_compliance/deploy_gitops.adoc]: Unfortunately I don't readily find a good resource, though if someone has gotten this far they've probably already been dealing with Openshift, so it might be a safe assumption that they're comfortable logging in to their cluster?

File: risk_compliance/deploy_gitops.adoc [hunk 8]
```diff
@@ -0,0 +1,177 @@
...
++
+View the following explanations for each argument:
+
+* URL: The URL to the repository that you forked from the main `policy-collection` repository. The default URL is `https://github.com/open-cluster-management/policy-collection.git`.
+
+* Branch: Branch of the Git repository to point to. The default branch is `main`.
+
+* Subdirectory Path: The subdirectory path you created to contain the policies you want to use. In the previous sample, we used the `my-policies` subdirectory, but you can also specify which folder you want start with. For example, you can use `my-policies/AC-Access-Control`. The default folder is `stable`.
+
+* Namespace: The namespace where the resources and policies are created on the hub cluster. These instructions use the `policy-namespace` namespace The default namespace is `policies`.
```
Comment (dhaiducek) [inline risk_compliance/deploy_gitops.adoc]: Missing a period:
>the `policy-namespace` namespace. The ...

File: risk_compliance/deploy_gitops.adoc [hunk 9]
```diff
@@ -0,0 +1,177 @@
+[#deploy-policies-using-gitops]
+= Deploy policies using GitOps
+
+The governance framework allows you to deploy a set of policies across a fleet of managed clusters. You can leverage the open source community, https://github.com/open-cluster-management/policy-collection[`policy-collection`] by contributing and using the policies in the repository. Policies in each of the `stable` and `community` folders from the open source community are further organized according to link:https://nvd.nist.gov/800-53/Rev4[NIST Special Publication 800-53].
```
Comment (swopebe) [inline risk_compliance/deploy_gitops.adoc]: Try to make first sentence active (avoiding anthropomorphic where you can), leverage I am not sure we usually use, prepositions here may be off "By contributing **to...**. and using policies **in**.

File: risk_compliance/deploy_gitops.adoc [hunk 10]
```diff
@@ -0,0 +1,177 @@
...
+Continue reading to learn how to use GitOps to automate and track policy updates and creation through a Git repository.
+
+*Prerequsite*: Before you begin, be sure to create you own fork of the `policy-collection` repository.
+
+[#customizing-your-repo]
+== Customizing your local repository
+
+Customize your local repository by consolidating the `stable` and `community` policies into a single folder. Remove the policies you do not want to use. Complete the following steps to customize your local repository:
+
+. Be sure that you are in your local `policy-collection` repository on your main default branch for GitOps. Create a new directory in the repository to hold the policies that you want to deploy. Run the following command:
```
Comment (swopebe) [inline risk_compliance/deploy_gitops.adoc]: Always start with the step itself, then the conditional or supplemental info. 

1. Create (the actual task).....   You can say: From your local repository....., create.....  User should know right after the step number that the task here is to **Create**.

File: risk_compliance/deploy_gitops.adoc [hunk 11]
```diff
@@ -0,0 +1,177 @@
...
+
++
+----
+mkdir my-policies
+----
+
+. Copy all of the `stable` and `community` policies into your `my-policies` directory. Start with the `community` policies first, in case the `stable` folder contains duplicates of what is available in the community. Run the following commands:
+
```
Comment (swopebe) [inline risk_compliance/deploy_gitops.adoc:23]: What you did here ^^^^ is exactly what I mean, beautiful!

File: risk_compliance/deploy_gitops.adoc [hunk 12]
```diff
@@ -0,0 +1,177 @@
...
++
+----
+cp -R community/* my-policies/
+
+cp -R stable/* my-policies/
+----
++
+Now that you have all of the policies in a single parent directory structure, you can make edits to the policies in your fork.
```
Comment (swopebe) [inline risk_compliance/deploy_gitops.adoc]: Concise here:

Instead of "you can make edits", just use the action word:

... you can edit.

**Review by dhaiducek — COMMENTED** [review]
Looks good to me, though "contributing policies" is mentioned in the title/link but never referenced. We should add a little blurb about contributing somewhere, maybe something like this (shamelessly stolen from our blog):
>As you develop and use policies to manage clusters with Red Hat Advanced Cluster Management, you can contribute the policies that you refine and create back to the `policy-collection` repository by submitting a Pull Request to broaden and strengthen our policy collection.

And then we could link to our contributing doc: https://github.com/open-cluster-management/policy-collection/blob/main/CONTRIBUTING.md#contributing-a-custom-policy

**Comment by dockerymick** [comment]
Thanks @dhaiducek and @swopebe for the review and comments!

File: risk_compliance/deploy_gitops.adoc [hunk 13]
```diff
@@ -0,0 +1,177 @@
...
++
+*Tips*:
+
+* It is recommend to remove the policies you are not planning to use.
+
+* Spend some time going through the policies to make sure you understand the following criteria for each policy:
```
Comment (swopebe) [inline risk_compliance/deploy_gitops.adoc]: Rather:

Learn about policies and the definition of the policies from the following list: 

(More concise and active)

File: risk_compliance/deploy_gitops.adoc (hunk 1 above)
Comment (swopebe) [inline risk_compliance/deploy_gitops.adoc]: missing code block here and maybe if you agree, emphasis on inform and enforce. Normally we say View or See the `spec.remediationAction`...

View the _Validate_ section.

File: risk_compliance/deploy_gitops.adoc (hunk 2 above)
Comment (swopebe) [inline risk_compliance/deploy_gitops.adoc]: code block on the label?

Red Hat OpenShift (we have a conref maybe use here)

Relative pronoun: Policy **that is** configured one way...

File: risk_compliance/deploy_gitops.adoc [hunk 14]
```diff
@@ -0,0 +1,177 @@
...
+
+** Placement: What clusters is the policy deployed to? By default, most policies target the clusters with the environment: dev label. Some policies may target OpenShift clusters or another label. You can update or add additional labels to include other clusters. When there is no specific value, the policy is applied to all of your clusters. You can also create multiple copies of a policy and customize each one if you want to use a policy configured one way for one set of clusters and configured another way for another set of clusters.
+
+[#committing-to-your-repo]
+== Committing to your local repository
+
+After you are satisfied with the changes you have made to your directory, commit and push your changes to Git so that they can be accessed by your cluster. 
+
+*Note*: This example is used to show the basics of how to use policies with GitOps, so you might have a different workflow to get changes to the desired branch of the repository.
+
```
Comment (swopebe) [inline risk_compliance/deploy_gitops.adoc:53]: I have a gitops example in app, as well. I suggest we both do the same thing when presenting the examples since they are not hard-lined examples. If you agree, we can look at both in the content call.
Here is my set up:
[#gitops-pattern]
== GitOps pattern

Learn best practices for organizing a Git repository to manage clusters.

[#repo-git-ops]
===  GitOps example

Comment (swopebe) [reply risk_compliance/deploy_gitops.adoc:53]: I don't think desired is an approved word, it wasn't previously.

File: risk_compliance/deploy_gitops.adoc [hunk 15]
```diff
@@ -0,0 +1,177 @@
...
+Complete the following steps:
+
+. From your terminal, run `git status` to view your recent changes in your directory that you previously created. Add your new directory to the list of changes to be committed with the following command:
++
+----
+git add my-policies/
+----
+
+. Commit the changes and customize the message as desired. Run the following command:
```
Comment (swopebe) [inline risk_compliance/deploy_gitops.adoc]: Can drop desired and just say "customize your message."

File: risk_compliance/deploy_gitops.adoc [hunk 16]
```diff
@@ -0,0 +1,177 @@
...
++
+----
+git commit -m “Policies to deploy to the hub cluster”
+----
+
+. Push the changes to the main default branch of your forked Git repository’s that is used for GitOps. Run the following command:
++
```
Comment (swopebe) [inline risk_compliance/deploy_gitops.adoc:69]: Never possessive. 

Push the changes to the main default branch of your forked repository that is used for GitOps.

File: risk_compliance/deploy_gitops.adoc [hunk 17]
```diff
@@ -0,0 +1,177 @@
...
+
+Your changes are committed.
+
+[#deploying-policies-to-your-cluster]
+== Deploying policies to your cluster
+
+After you push your changes, you can deploy the policies to your {product-title} installation. Post deployment, your hub cluster is conncted to your Git repository. Any further changes to your chosen branch of the Git repository is reflected in your cluster.
+
+The `deploy.sh` script creates `Channel` and `Subscription` resources in your hub cluster. The channel connects to the Git repository, and the subscription specifies the data to bring to the cluster through the channel. In this way, all policies defined in the specified subdirectory is created on your hub. After the policies are created by the subscription, {product-title-short} analyzes the policies and creates additional policy resources in the namespace associated with each managed cluster that the policy is applied to, based on the defined placement rule.
+
```
Comment (swopebe) [inline risk_compliance/deploy_gitops.adoc:82]: In this way

change to 

As a result, (or something)

Furthermore, subject verb disagreement in the middle sentence. Two spots need the relative pronouns.

File: risk_compliance/deploy_gitops.adoc [hunk 18]
```diff
@@ -0,0 +1,177 @@
...
+The policy is then copied down to the managed cluster from its respective managed cluster namespace on the hub. In that way, the policies in your Git repository are pushed to all managed clusters that have labels that match the `clusterSelector` defined in the placement rule of your policy.
```
Comment (swopebe) [inline risk_compliance/deploy_gitops.adoc]: Same thing with: In that way.

Get rid of copied down, just copied; hub _cluster_ ; _that are_ defined.

File: risk_compliance/deploy_gitops.adoc [hunk 19]
```diff
@@ -0,0 +1,177 @@
...
+----
+oc cluster-info
+----
++
+The output of the command displays the API server details for the cluster, where {product-title-short} is installed. If the correct URL is not displayed, configure your CLI to point to the correct cluster. 
+
+//^is there a reference that we can add to help the
+// user configure the CLI?
+
+. Create a namespace where your policies are created to control access and identify them readily. Run the following command:
```
Comment (swopebe) [inline risk_compliance/deploy_gitops.adoc]: Consider rewriting this line, as I read it:

Create a namespace where your policies are created to control access. (try also not to write "them")

File: risk_compliance/deploy_gitops.adoc [hunk 20]
```diff
@@ -0,0 +1,177 @@
...
+----
+oc create namespace policy-namespace
+----
+
+. Run the following command to deploy the policies to your cluster:
++
+----
+./deploy.sh -u https://github.com/<your-repository>/policy-collection -p my-policies -n policy-namespace
+----
++
```
Comment (swopebe) [inline risk_compliance/deploy_gitops.adoc:114]: Looks like code blocks may be off here.

File: risk_compliance/deploy_gitops.adoc [hunk 21]
```diff
@@ -0,0 +1,177 @@
...
+Replace `your-repository` with your Git username or repository name.
++
+*Note*: For reference, the full list of arguments for the deploy.sh script uses the following syntax:
```
Comment (swopebe) [inline risk_compliance/deploy_gitops.adoc]: `deploy.sh`

File: risk_compliance/deploy_gitops.adoc [hunk 22]
```diff
@@ -0,0 +1,177 @@
...
+----
+
+. Run the following command to deploy the policies to your cluster:
++
+----
+./deploy.sh -u https://github.com/<your-repository>/policy-collection -p my-policies -n policy-namespace
+----
++
+Replace `your-repository` with your Git username or repository name.
++
```
Comment (swopebe) [inline risk_compliance/deploy_gitops.adoc:116]: user name

File: risk_compliance/deploy_gitops.adoc [hunk 23]
```diff
@@ -0,0 +1,177 @@
...
+----
+oc label policy <policy-name> -n <policy-namespace> <key>=<value>
+----
++
+Then, you can query policies that have labels with the following command:
++
+----
+oc get policies -n <policy-namespace> -l <key>=<value>
+----
+
```
Comment (swopebe) [inline risk_compliance/deploy_gitops.adoc:176]: Can they verify this?

File: risk_compliance/deploy_gitops.adoc [hunk 24]
```diff
@@ -0,0 +1,177 @@
...
++
+----
+apps.open-cluster-management.io/hosting-deployable: policies/deploy-stable-policies-Policy-policy-role9
+
+apps.open-cluster-management.io/hosting-subscription: policies/demo-policies
+
+apps.open-cluster-management.io/sync-source: subgbk8s-policies/demo-policies
+----
++
+GitOps annotations are valuable to see which subscription created the policy. You can also add your own labels to your policies, so that you can write runtime queries that select policies based on labels.
```
Comment (swopebe) [inline risk_compliance/deploy_gitops.adoc]: no comma here, can remove.

File: risk_compliance/deploy_gitops.adoc [hunk 25]
```diff
@@ -0,0 +1,177 @@
...
+* Branch: Branch of the Git repository to point to. The default branch is `main`.
+
+* Subdirectory Path: The subdirectory path you created to contain the policies you want to use. In the previous sample, we used the `my-policies` subdirectory, but you can also specify which folder you want start with. For example, you can use `my-policies/AC-Access-Control`. The default folder is `stable`.
+
+* Namespace: The namespace where the resources and policies are created on the hub cluster. These instructions use the `policy-namespace` namespace The default namespace is `policies`.
+
+* Name Prefix: Prefix for the `Channel` and `Subscription` resources. The default is `demo-stable-policies`.
+
+After you run the `deploy.sh` script, any user with access to the repository can commit changes to the branch, which pushes changes to exisitng policies on your clusters.
+
```
Comment (swopebe) [inline risk_compliance/deploy_gitops.adoc:136]: existing

File: risk_compliance/deploy_gitops.adoc [hunk 26]
```diff
@@ -0,0 +1,177 @@
...
+== Verifying GitOps policy deployments from the console
+
+Verify that your changes were applied to your policies from the console. You can also make more changes to your policy from the console. Complete the following steps:
+
+. Log in to your {product-title-short} cluster.
+
+. From the navigation menu, select *Risk and compliance*.
+
+. Consider the following questions:
++
```
Comment (swopebe) [inline risk_compliance/deploy_gitops.adoc:147]: This is not really a step. "Consider" because you are not telling them to do anything. We don't usually present questions in tech doc unless we do an FAQ blog or something. I would reword this and create a task for the user.

Similar to:

Check for the following policy details (or something like that).

File: risk_compliance/deploy_gitops.adoc [hunk 27]
```diff
@@ -0,0 +1,177 @@
...
+** Why is a specific policy compliant or non-compliant on the clusters that it was distributed to?
+** Are the policies applied to the correct clusters?
+** If this policy is not distributed to any clusters, why?
+. Identify the GitOps deployed policies that you created or modified. The GitOps deployed policies can be identified by the annotation that is applied automatically. Annotations for the GitOps deployed policies resemble the following paths:
++
+----
```
Comment (swopebe) [inline risk_compliance/deploy_gitops.adoc:153]: code seems off here

**Review by swopebe — CHANGES_REQUESTED** [review]
See comments, questions, recommendations.

File: risk_compliance/deploy_gitops.adoc (hunk 7 above)
Comment (dockerymick) [reply risk_compliance/deploy_gitops.adoc]: I took a look in the OCP docs. @dhaiducek would this link be an accurate reference: https://docs.openshift.com/container-platform/4.7/cli_reference/openshift_cli/getting-started-cli.html#cli-using-cli_cli-developer-commands??

File: risk_compliance/deploy_gitops.adoc (hunk 20 above)
Comment (dockerymick) [reply risk_compliance/deploy_gitops.adoc:114]: From the Preview tabs, the alignment looks ok from my end

File: risk_compliance/deploy_gitops.adoc (hunk 27 above)
Comment (dockerymick) [reply risk_compliance/deploy_gitops.adoc:153]: These are directory paths, should i just separate with a comma instead of using the codeblock?

**Comment by dockerymick** [comment]
> Looks good to me, though "contributing policies" is mentioned in the title/link but never referenced. We should add a little blurb about contributing somewhere, maybe something like this (shamelessly stolen from our blog):
> 
> > As you develop and use policies to manage clusters with Red Hat Advanced Cluster Management, you can contribute the policies that you refine and create back to the `policy-collection` repository by submitting a Pull Request to broaden and strengthen our policy collection.
> 
> And then we could link to our contributing doc: https://github.com/open-cluster-management/policy-collection/blob/main/CONTRIBUTING.md#contributing-a-custom-policy

The title for this file is titled Deploy policies using GitOps. I do mention that the user can contribute to the policy, so I will add the reference link to the contributing guidelines

File: risk_compliance/deploy_gitops.adoc (hunk 14 above)
Comment (dockerymick) [reply risk_compliance/deploy_gitops.adoc:53]: Thanks for the suggestion @swopebe I took a look at your section. I like your descriptions of the how the directory is set up and how it should flow, however I don't agree with adding this info into the deploy_gitops file. I do like the intro statement that you use, my change might resemble the following statement: "Learn best practice for organizing a Git repository to deploy and manage your policies."

Comment (swopebe) [reply risk_compliance/deploy_gitops.adoc:53]: That sounds great.

**Review by swopebe — APPROVED** [review]
Looks great, M!

**Comment by dockerymick** [comment]
Thank you for the review @swopebe @dhaiducek  ! I appreciate you both. Moving forward with the merge
